RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc

```diff
@@ -301,12 +335,22 @@ void ReducedEGProducer::produce(edm::Event& theEvent, const edm::EventSetup& the
   for (const auto& photon : *photonHandle) {
     index++;
 
-    bool keep = keepPhotonSel_(photon);
-    if (!keep) continue;
-    
     reco::PhotonRef photonref(photonHandle,index);
-    
     photons->push_back(photon);
+    auto& newPhoton = photons->back();
+   
+    if( (applyPhotonCalibOnData_ && theEvent.isRealData()) ||
+	(applyPhotonCalibOnMC_ && !theEvent.isRealData()) ){
+      calibratePhoton(newPhoton,photonref,		      
+		      *photonCalibEnergyHandle,*photonCalibEnergyErrHandle);
+    }
+      
+    //we do this after calibration
+    bool keep = keepPhotonSel_(newPhoton);
+    if (!keep){
+      photons->pop_back();
+      continue;
+    }    
```

Comment:
shouldn't the same machinery be made available for ootPhotons as well?

Reply:
In theory yes, but I decided that OOT should run their own energy corrections to simplify things @kmcdermo

Reply:
I agree with Sam's point here.  Any energy corrections for OOT would come from us separately, perhaps following the same routine with a different config at some point.  To first order, we are not interested in precise energy resolution. For now, we can be left alone to speed this approval.